Intel_SighPP_CMake/Intel_SighPP_CMake/src/interface_controller/output_stream_controller.cpp

```diff
@@ -41,72 +41,88 @@ void OutputStreamController::output_to_depth_window(const rs2::frame& frame) con
+
+
+	for (auto&& item : vector)
 	{
-		for (auto&& item : prioritised_classification_result.objects)
-		{
-			cv::Rect object(item.bottom_left.x, item.bottom_left.y, item.top_right.x - item.bottom_left.x, item.top_right.y - item.bottom_left.y);
-			object = object & cv::Rect(0, 0, depth_matrix.cols, depth_matrix.rows);
+		cv::Rect object(item.bottom_left.x, item.bottom_left.y, item.top_right.x - item.bottom_left.x, item.top_right.y - item.bottom_left.y);
+		object = object & cv::Rect(0, 0, depth_matrix.cols, depth_matrix.rows);
 
-			std::ostringstream ss;
-			ss << item.name << " ";
-			ss << std::setprecision(2) << item.distance << " meters away";
-			cv::String conf(ss.str());
+		std::ostringstream ss;
+		ss << item.name << ", ID:";
+		ss << item.id << ", conf:";
+		ss << item.confidence << " ";
+		ss << std::setprecision(2) << item.distance << " meters away";
+		cv::String conf(ss.str());
 
-			rectangle(color_matrix, object, cv::Scalar(0, 255, 0));
-			auto base_line = 0;
-			auto label_size = cv::getTextSize(ss.str(), cv::FONT_HERSHEY_SIMPLEX, 0.5, 1, &base_line);
+		rectangle(color_matrix, object, cv::Scalar(0, 255, 0));
+		int base_line = 0;
+		cv::Size label_size = cv::getTextSize(ss.str(), cv::FONT_HERSHEY_SIMPLEX, 0.5, 1, &base_line);
 
-			auto center = (object.br() + object.tl()) * 0.5;
-			center.x = center.x - label_size.width / 2;
+		auto center = (object.br() + object.tl()) * 0.5;
+		center.x = center.x - label_size.width / 2;
 
-			rectangle(
-				color_matrix,
-				cv::Rect(cv::Point(center.x, center.y - label_size.height),
-					cv::Size(label_size.width, label_size.height + base_line)),
-				cv::Scalar(255, 255, 255), cv::FILLED);
+		rectangle(
+			color_matrix,
+			cv::Rect(cv::Point(center.x, center.y - label_size.height),
+				cv::Size(label_size.width, label_size.height + base_line)),
+			cv::Scalar(255, 255, 255), cv::FILLED);
+
+		cv::putText(color_matrix, ss.str(), center, cv::FONT_HERSHEY_SIMPLEX, 0.5, cv::Scalar(0, 0, 0));
 
-			putText(color_matrix, ss.str(), center, cv::FONT_HERSHEY_SIMPLEX, 0.5, cv::Scalar(0, 0, 0));
-		}
 	}
 
 	// Switch RGB to BGR as openCV uses BGR
-	cv::Mat bgr;
-	cvtColor(color_matrix, bgr, cv::COLOR_RGB2BGR);
-	cv::imshow(color_output_window_, bgr);
+	//cv::Mat bgr;
+	//cvtColor(color_matrix, bgr, cv::COLOR_RGB2BGR);
+	cv::imshow(color_output_window_, color_matrix);
 }
 
-bool OutputStreamController::should_receive_new_frames() const
-{
-	return wait_key() && is_depth_window_ready() && is_color_window_ready();
-}
+	/// <summary>
+	/// Tells the system if the opencv stream output windows are ready, if enabled.
+	/// In the case where no output stream windows are enabled, returns true.
+	/// Based on the rs-dnn and rs-imshow examples.
+	/// </summary>
+	/// <returns>
+	///   Returns TRUE if the windows can receive frames, or if no output stream windows are enabled.
+	///   Returns FALSE if the windows can not receive frames.
+	/// </returns>
+
```

Comment:
The summary exists in the hpp file

Reply:
fair

Reply:
Yeah, it should be in the header file, not the source file 😄